flake.nix

```diff
@@ -5,99 +5,129 @@
     # np.url = "github:nixos/nixpkgs?ref=haskell-updates";
     fu.url = "github:numtide/flake-utils?ref=master";
     ds.url = "github:numtide/devshell?ref=master";
+    nf.url = "github:numtide/nix-filter?ref=master";
     hls.url = "github:haskell/haskell-language-server?ref=master";
     shs.url =
       "github:luc-tielen/souffle-haskell?rev=c46d0677e4bc830df89ec1de2396c562eb9d86d3";
-    llvm-cg.url =
-      "github:luc-tielen/llvm-codegen?rev=83306c9da879b80be674ef7cf910aaf158160b69";
+    llvm-cg.url = "github:luc-tielen/llvm-codegen";
+    llvm-cg.flake = false;
     alga.url =
       "github:snowleopard/alga?rev=75de41a4323ab9e58ca49dbd78b77f307b189795";
     alga.flake = false;
-    diagnose.url = "github:luc-tielen/diagnose?rev=d58752f062c105ec0f8831357f3c688965e13add";
+    diagnose.url =
+      "github:luc-tielen/diagnose?rev=d58752f062c105ec0f8831357f3c688965e13add";
     diagnose.flake = false;
   };
-  outputs = { self, fu, ds, shs, llvm-cg, ... }@inputs:
+  outputs = { self, fu, nf, ds, shs, llvm-cg, ... }@inputs:
     let
       # Use the same nixpkgs as souffle-haskell, to avoid weird issues with multiple versions of Haskell packages.
       np = shs.inputs.np;
-    in
-      with np.lib;
-      with fu.lib;
-      eachSystem [ "x86_64-linux" ] (system:
-        let
-          ghcVersion = "902";
-          version = "${ghcVersion}.${substring 0 8 self.lastModifiedDate}.${
-              self.shortRev or "dirty"
-            }";
-          config = {};
-          overlay = final: _:
-            let
-              haskellPackages =
-                final.haskell.packages."ghc${ghcVersion}".override {
-                  overrides = hf: hp:
-                    with final.haskell.lib; rec {
-                      inherit (shs.packages."${system}") souffle-haskell;
-                      inherit (llvm-cg.packages."${system}") llvm-codegen;
+    in with np.lib;
+    with fu.lib;
+    eachSystem [ "x86_64-linux" ] (system:
```

Comment:
I think we still need to add x86_64-darwin to actually make it run on that arch.

Reply:
Yes, that's the next step indeed. And that's part of the reason why I'm removing the llvm-cg overlay. @placidex has successfully tested it in a separate branch

Reply:
Ok, didn't realize that!